src/container/cli.c

```diff
@@ -1927,8 +1952,9 @@ dc_cont_g2l(daos_handle_t poh, struct dc_cont_glob *cont_glob,
 	D_RWLOCK_WRLOCK(&pool->dp_co_list_lock);
 	if (pool->dp_disconnecting) {
 		D_RWLOCK_UNLOCK(&pool->dp_co_list_lock);
+		dc_cont_free(cont);
 		D_ERROR("pool connection being invalidated\n");
-		D_GOTO(out_cont, rc = -DER_NO_HDL);
+		D_GOTO(out_pool, rc = -DER_NO_HDL);
```

Comment:
unnecessary change, goto out_cont is to free the dc_cont allocated above L1944, that is same as you changed dc_cont_free + goto out_pool

Reply:
Agree. But this is better I think. If you insist I will restore original one.